src/Persistence.Linnapps/Repositories/VatCodeRepository.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace Linn.Products.Persistence.Linnapps.Repositories
+{
+    using System;
+    using System.Linq;
+    using System.Linq.Expressions;
+
+    using Linn.Common.Persistence;
+    using Linn.Products.Domain.Linnapps;
+
+    public class VatCodeRepository : IRepository<VatCode, string>
+    {
+        private readonly ServiceDbContext serviceDbContext;
+
+        public VatCodeRepository(ServiceDbContext serviceDbContext)
+        {
+            this.serviceDbContext = serviceDbContext;
+        }
+
+        public VatCode FindById(string key)
+        {
+            return this.serviceDbContext.VatCodes.Where(a => a.Code == key).ToList().First();
+        }
+
+        public IQueryable<VatCode> FindAll()
+        {
+            return this.serviceDbContext.VatCodes;
+        }
+
+        public void Add(VatCode vatCode)
+        {
+            this.serviceDbContext.VatCodes.Add(vatCode);
+            this.serviceDbContext.SaveChanges();
```

Comment:
You shouldn't need the save changes in here. 
It's not really the repository's business to meddle with the transaction (and the facade service does it).